From e83ea84afecc74e1c1e96b8b4f9042ca3659be0a Mon Sep 17 00:00:00 2001
From: Guy Harris <guy@alum.mit.edu>
Date: Tue, 27 Aug 2013 10:59:31 -0700
Subject: [PATCH] Fix (unlikely) memory leak.

If we were able to allocate the list of time stamp types, but were
unable to allocate the list of time stamp precisions, free the list of
time stamp types before freeing the handle.
---
 pcap-linux.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/pcap-linux.c b/pcap-linux.c
index 5856dc4..cd3131a 100644
--- a/pcap-linux.c
+++ b/pcap-linux.c
@@ -427,6 +427,8 @@ pcap_create_interface(const char *device, char *ebuf)
 	handle->tstamp_precision_count = 2;
 	handle->tstamp_precision_list = malloc(2 * sizeof(u_int));
 	if (handle->tstamp_precision_list == NULL) {
+		if (handle->tstamp_type_list != NULL)
+			free(handle->tstamp_type_list);
 		free(handle);
 		return NULL;
 	}
-- 
1.8.3.1

